src/Microsoft.Azure.WebJobs.Host/Blobs/BlobPath.cs

```diff
@@ -55,16 +56,30 @@ public static BlobPath ParseAndValidate(string value, bool isContainerBinding =
             return path;
         }
 
-        public static BlobPath Parse(string value, bool isContainerBinding)
+        // throws exception if failed
+        public static BlobPath ParseAbsUrl(string blobUrl)
```

Comment:
https://github.com/Azure/azure-webjobs-sdk/blob/e692a1ff0db6decf1ba216752c7d86414d4489ca/src/Microsoft.Azure.WebJobs.Host/Blobs/Listeners/StorageAnalyticsLogEntry.cs#L131-L136
can be replace by this to use storage sdk for parsing, but I didn't change it since it also parse the nonAbsoluteUrl..